### ISSUE TITLE
Question about Schedule

### ISSUE THREAD

**Issue by DoTulip** [deepmind/optax#424]
Hello Optax team,
I am a new user of optax and have some questions about using it. How to use schedule when using adam? I found that when using adam, the learning rate is constant even when the schedule is passed in. Can only adamw be used with schedule? Is it possible to add some examples on how to use schedule?

**Comment by DoTulip** [comment]
I did it this way, but I found that the learning rate didn't change at all.
```
init_value, peak_value, end_value, total_steps, warmup_steps = 1e-6, 1e-3, 1e-6, 1e4, 2e3
schedule_fn = optax.warmup_cosine_decay_schedule(init_value, peak_value, warmup_steps, total_steps, end_value)
scheduled_adam = optax.inject_hyperparams(optax.adam)(learning_rate=schedule_fn)

def train_step(params, state):
    (loss_val, batch_stats), grads = loss_grad_fn(params)
    updates_params, opt_state = optimizer.update(grads, state, params)
    params = optax.apply_updates(params, updates_params)
    return params, loss_val, batch_stats, opt_state

params, loss_val, batch_stats, opt_state = train_step(params, state) 
print(opt_state.hyperparams['learning_rate'])
```
Looking forward to your help！ @mkunesch @n2cholas @rosshemsley @mtthss

**Comment by DoTulip** [comment]
Sorry to interrupt. I made a stupid mistake when calling the schedule, which has been solved now! I will close this issue. Thank you again for creating such a powerful tool!

**Comment by rosshemsley** [comment]
No worries @DoTulip. If it's possible your mistake could be made by others, please do feel free to share what the mistake was so that others can learn from it too :)
